File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_line'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Line Bricklet

ipcon = IPConnection.new # Create IP connection
l = BrickletLine.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register reflectivity callback
l.register_callback(BrickletLine::CALLBACK_REFLECTIVITY) do |reflectivity|
  puts "Reflectivity: #{reflectivity}"
end

# Set period for reflectivity callback to 1s (1000ms)
# Note: The reflectivity callback is only called every second
#       if the reflectivity has changed since the last call!
l.set_reflectivity_callback_period 1000

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'j6q'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var line = new Tinkerforge.BrickletLine(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set Period for reflectivity callback to 1s (1000ms)
        // Note: The reflectivity callback is only called every second if the
        // reflectivity has changed since the last call!
        line.setReflectivityCallbackPeriod(1000);
    }
);

// Register reflectivity callback
line.on(Tinkerforge.BrickletLine.CALLBACK_REFLECTIVITY,
    // Callback function for reflectivity
    function(reflectivity) {
        console.log('Reflectivity: '+reflectivity);
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
